Networker/Common/PacketHandlerDecorator.cs

```diff
@@ -0,0 +1,55 @@
+﻿using Networker.Common.Abstractions;
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Threading.Tasks;
+
+namespace Networker.Common
+{
+    public abstract class PacketHandlerDecorator<T> : PacketHandlerBase<T> where T: class
```

Comment:
This particular way of doing it would work well as an extension project, but I would like to avoid the dependency on PacketHandlerBase<T> as you are able to use just an IPacketHandler now.

Would this also mean you'd need a new PacketHandlerDecorator for each packet type?

Reply:
Yeah implementing IPacketHandler makes more sense. The decorator has helper methods to get the target handler method info based on the packet type. This assumes the handler methods accept a base packet type thought. I definitely need to think more about this.

Where/how should I wire up the decorators? Once I get it to a testable state, I can play around with ideas to make sure a single decorator can handle multiple packet types.

Reply:
I've got a new way of registering packets + handlers coming in soon which might be the best place to wire up the decorator, I'll let you know when it's ready.

Reply:
Alright sounds good!